internal/app/init/pkg/system/services/osd.go

```diff
@@ -82,3 +86,26 @@ func (o *OSD) Runner(data *userdata.UserData) (runner.Runner, error) {
 		restart.WithType(restart.Forever),
 	), nil
 }
+
+// HealthFunc implements the HealthcheckedService interface
+func (o *OSD) HealthFunc(*userdata.UserData) health.Check {
+	return func(ctx context.Context) error {
+		var d net.Dialer
+		conn, err := d.DialContext(ctx, "tcp", fmt.Sprintf("%s:%d", "localhost", constants.OsdPort))
```

Comment:
Just keep in mind that the only thing we get from this is a historical reference. Not saying that isn't worth it but the health of OSD can't be retrieved if its down. But this can be super useful to see if it _was_ down at some point and has come back up since.

Reply:
exactly, init will be running no matter what and capturing events, so we can see if system is e.g. super-busy and osd doesn't respond in time